### PR TITLE
Roman/bugfix conflicting event loop ingest

### PR DESCRIPTION
**Pull request by rbiseck3**
### Description
In use cases where an external system (such as code being run in a jupyter notebook) already has a running event loop, run the async code in a dedicated thread pool to not conflict with the existing event loop.

This also has a variety of fixes that were found when putting together a demo leveraging the elasticsearch destination connector

### CODE REVIEW

File: scripts/elasticsearch-test-helpers/destination_connector/test-ingest-elasticsearch-output.py [hunk 1]
```diff
@@ -39,32 +38,96 @@ def query(client: Elasticsearch, search_text: str):
...
+    print(f"Validating that the count of documents in index {INDEX_NAME} is {num_elements}")
     count = int(client.cat.count(index=INDEX_NAME, format="json")[0]["count"])
-    try:
-        assert count == N_ELEMENTS
-    except AssertionError:
-        sys.exit(
-            "Elasticsearch dest check failed:"
-            f"got {count} items in index, expected {N_ELEMENTS} items in index."
-        )
+    consistent = False
```
Comment (potter-potter) [inline scripts/elasticsearch-test-helpers/destination_connector/test-ingest-elasticsearch-output.py:44]: What was the motivation for this? We were getting different counts?

File: unstructured/ingest/v2/pipeline/steps/upload.py [hunk 2]
```diff
@@ -44,18 +44,12 @@ def __post_init__(self):
...
-        logger.info("processing content async")
-        return asyncio.run(self._process_async(iterable=iterable))
-
+    @timed
     def __call__(self, iterable: iterable_input):
         logger.info(
             f"Calling {self.__class__.__name__} " f"with {len(iterable)} docs",  # type: ignore
         )
-        if self.process.is_async():
+        if self.process.is_async() and self.process.is_async():
```
Comment (potter-potter) [inline unstructured/ingest/v2/pipeline/steps/upload.py:52]: This seems strange. why test for is_async twice?

File: scripts/elasticsearch-test-helpers/destination_connector/test-ingest-elasticsearch-output.py (hunk 1 above)
Comment (rbiseck3) [reply scripts/elasticsearch-test-helpers/destination_connector/test-ingest-elasticsearch-output.py:44]: If there's anything surrounding chunking that affects this, it is no longer a static count. I did find that this count changed based on fixes to things like the chunker but this test is meant to isolate only the destination connector. These changes made it a bit more dynamic and based on the results that came out of the stager step in the pipeline.

File: unstructured/ingest/v2/pipeline/steps/upload.py (hunk 2 above)
Comment (rbiseck3) [reply unstructured/ingest/v2/pipeline/steps/upload.py:52]: Good catch, must have been a copy-paste error.

File: unstructured/ingest/v2/processes/partitioner.py [hunk 3]
```diff
@@ -116,14 +116,25 @@ def create_partition_parameters(self, filename: Path) -> "PartitionParameters":
         from unstructured_client.models.shared import Files, PartitionParameters
 
         partition_request = self.config.to_partition_kwargs()
+        possible_fields = [f.name for f in fields(PartitionParameters)]
+        filtered_partition_request = {
+            k: v for k, v in partition_request.items() if k in possible_fields
+        }
+        if len(filtered_partition_request) == len(partition_request):
```
Comment (potter-potter) [inline unstructured/ingest/v2/processes/partitioner.py:123]: I don't understand this. I ran a test with 
--strategy fast \
--partition-by-api \

And got this:
`Following fields were omitted due to not being supported by the currently used unstructured client: `
with nothing past the `:`

Comment (rbiseck3) [reply unstructured/ingest/v2/processes/partitioner.py]: Had the boolean logic wrong, updated it from `==` to `!=` so this shouldn't show up anymore if no fields were omitted.

**Review by potter-potter — APPROVED** [review]
works well

**Pull request merged**